api/bin/chainflip-ingress-egress-tracker/src/witnessing/state_chain.rs

```diff
@@ -122,95 +185,256 @@ impl Storable for WitnessInformation {
+					input_asset: self.input_asset.into(),
+					output_asset: self.output_asset,
+					amount: self.deposit_amount.into(),
+					destination_address: destination_address_from_encoded_address(
+						self.destination_address,
+						network,
+					)?,
+					ccm_deposit_metadata: self.deposit_metadata,
+					deposit_details: self.deposit_details.into_any_chain(),
+					broker_fee: self.broker_fee.clone(),
+					affiliate_fees: map_affiliates(
+						state_chain_client,
+						self.affiliate_fees,
+						self.broker_fee.account,
+					)
+					.await,
+					refund_params: self.refund_params,
+					dca_params: self.dca_params,
+					boost_fee: self.boost_fee,
+				})
+			}
 		}
+	};
+}
+
+impl_vault_deposit_into_witness_information!(Ethereum);
+impl_vault_deposit_into_witness_information!(Bitcoin);
+impl_vault_deposit_into_witness_information!(Polkadot);
+impl_vault_deposit_into_witness_information!(Arbitrum);
+
+impl BroadcastIntoWitnessInformation for BroadcastDetails<Ethereum> {
+	fn into_witness_information(self) -> anyhow::Result<WitnessInformation> {
+		Ok(WitnessInformation::Broadcast {
+			broadcast_id: self.broadcast_id,
+			tx_out_id: TransactionId::Ethereum { signature: self.tx_out_id },
+			tx_ref: TransactionRef::Ethereum { hash: self.tx_ref },
+		})
 	}
 }
 
-impl From<DepositInfo<Arbitrum>> for WitnessInformation {
-	fn from((value, height, _): DepositInfo<Arbitrum>) -> Self {
-		Self::Deposit {
-			deposit_chain_block_height: height,
-			deposit_address: hex_encode_bytes(value.deposit_address.as_bytes()),
-			amount: value.amount.into(),
-			asset: value.asset.into(),
-			deposit_details: value
-				.deposit_details
-				.tx_hashes
-				.map(|tx_hashes| DepositDetails::Arbitrum { tx_hashes }),
-		}
+impl BroadcastIntoWitnessInformation for BroadcastDetails<Arbitrum> {
+	fn into_witness_information(self) -> anyhow::Result<WitnessInformation> {
+		Ok(WitnessInformation::Broadcast {
+			broadcast_id: self.broadcast_id,
+			tx_out_id: TransactionId::Arbitrum { signature: self.tx_out_id },
+			tx_ref: TransactionRef::Arbitrum { hash: self.tx_ref },
+		})
 	}
 }
 
-async fn save_deposit_witnesses<S: Store, C: Chain>(
+impl BroadcastIntoWitnessInformation for BroadcastDetails<Bitcoin> {
+	fn into_witness_information(self) -> anyhow::Result<WitnessInformation> {
+		Ok(WitnessInformation::Broadcast {
+			broadcast_id: self.broadcast_id,
+			tx_out_id: TransactionId::Bitcoin { hash: BitcoinHash(self.tx_out_id) },
+			tx_ref: TransactionRef::Bitcoin { hash: BitcoinHash(self.tx_ref) },
+		})
+	}
+}
+
+impl BroadcastIntoWitnessInformation for BroadcastDetails<Polkadot> {
+	fn into_witness_information(self) -> anyhow::Result<WitnessInformation> {
+		Ok(WitnessInformation::Broadcast {
+			broadcast_id: self.broadcast_id,
+			tx_out_id: TransactionId::Polkadot {
+				signature: DotSignature(*self.tx_out_id.aliased_ref()),
+			},
+			tx_ref: TransactionRef::Polkadot { transaction_id: self.tx_ref },
+		})
+	}
+}
+
+async fn save_deposit_witnesses<
```

Comment:
I feel like we could simplify a lot if we query for a map of affiliate IDs in this method, and then pass that through, rather than threading the sc client (and attendant type constraints and async-ness) down through the whole call stack.